### ISSUE TITLE
[flaky test][arm builds] CI/CD failing during arm builds intermittently

### ISSUE THREAD

**Issue by crobert-1** [open-telemetry/opentelemetry-collector-contrib#32286]
### Describe the issue you're reporting

[Failure link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8622302341/job/23633699812?pr=32221)

```
Run actions/setup-go@v5
  with:
    go-version: 1.[2](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8622302341/job/23633699812?pr=32221#step:6:2)2.0
    cache: false
    check-latest: false
    token: ***
  env:
    TEST_RESULTS: testbed/tests/results/junit/results.xml
    SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2
    PATH: /home/runner/.arkade/bin:/home/runner/.dotnet/tools:/home/runner/.cargo/bin:/home/runner/go/bin:/home/runner/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
Setup go version spec 1.22.0
Attempting to download 1.22.0...
matching 1.22.0...
Not found in manifest.  Falling back to download directly from Go
Error: Request timeout: /dl/?mode=json&include=all
```

**Comment by crobert-1** [comment]
It looks like a potential fix may be to upgrade to use `1.22.2` for CI/CD, as the [`go-version` manifest](https://github.com/actions/go-versions/blob/497900d98518d9413199ddd87817399f19733d92/versions-manifest.json#L3) includes ARM based builds for Linux, Darwin, and Windows. `1.22.0` only has an ARM build for Darwin.

**Comment by crobert-1** [comment]
This wouldn't explain why it's succeeding sometimes though, so this may not resolve it. I've opened a draft PR to see if that fixes it.
